### PR TITLE
Design: US-2.8 Company Search

### PR DESCRIPTION
**Pull request by galexy**
## Summary

Design documents for **US-2.8: Company Search** (beads: `edgar-diff-u2l`).

- **Implementation Design** (`.specs/us-2.8-company-search/implementation-design.md`): Architecture for transforming the placeholder SearchBar into a functional company search. Key decisions:
  - Vite dev proxy for CORS (no new deps)
  - Bundled `company_tickers.json` for fast ticker→CIK resolution
  - Custom `useCompanySearch` hook (no Redux/context)
  - Custom `useDebouncedValue` hook (zero deps, 300ms)
  - Two-stage resolution: local ticker lookup → SEC submissions API

- **Test Plan** (`.specs/us-2.8-company-search/test-plan.md`): Comprehensive BDD scenarios, unit/integration/e2e test coverage, boundary conditions, error handling, and UAT scenarios.

Both documents were collaboratively reviewed and aligned between coder and tester agents.

## Beads Issue
`edgar-diff-u2l` — US-2.8: Company Search

## Review Checklist
- [ ] Implementation approach is sound
- [ ] Architecture decisions are reasonable
- [ ] Test coverage is comprehensive
- [ ] Edge cases are addressed

Co-Authored-By: SageOx <ox@sageox.ai>

### CODE REVIEW

File: .specs/us-2.8-company-search/implementation-design.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+
+## Overview
+
+Transform the disabled placeholder `SearchBar` into a functional company search that resolves tickers, names, and CIKs against SEC EDGAR data, displaying the resolved company as confirmation.
+
+---
+
+## Approach
+
+### 1. CORS Proxy: Vite Dev Server Proxy
```
Comment (galexy) [inline .specs/us-2.8-company-search/implementation-design.md]: Can we use Cloudflare workers for both #1 and #2? If that is the case, can use the wrangler to create the workers and they run locally for dev? There should also be a vite plugin.

**Review by galexy — COMMENTED** [review]
UAT test plan is missing.

**Comment by galexy** [comment]
[Tester Agent] Added comprehensive UAT test plan section. Updated integration/e2e tests to align with Cloudflare Workers architecture.

**UAT doc:** `.specs/us-2.8-company-search/uat.md` — 17 test steps covering:
- Visual layout (search bar appearance, focus state, dropdown)
- Interactions (hover, keyboard navigation, selection, sequential search)
- Loading, error, and empty states
- Responsive behavior (768px and 375px viewports)
- Accessibility (ARIA combobox pattern verification)
- Worker endpoint verification (CORS, routing)

**Test plan updates for Cloudflare Workers:**
- Added section 2.6: Worker unit tests (route matching, CORS headers, tickers caching, submissions proxy)
- Updated company-resolver tests: `/data/company_tickers.json` → `/api/tickers`
- Replaced Vite proxy section with Worker integration checks
- Added Worker 502 and tickers API failure error conditions
- Updated coverage matrix with Worker, responsive, and accessibility rows
